### ISSUE TITLE
Imported Interface falsely detected as unused

### ISSUE THREAD

**Issue by simonhaenisch** [palantir/tslint#3307]
### Bug Report

- **TSLint version:** `5.7.0`
- **TypeScript version:** `2.3.4`
- **Running TSLint via:** CLI (Ionic)

#### TypeScript code being linted

```ts
import { Injectable } from '@angular/core';
import { Apollo } from 'apollo-angular';
import { MyInterface } from './interfaces';

@Injectable()
export class Api {
  constructor(public apollo: Apollo) {}

  query() {
    return this.apollo.watchQuery<MyInterface>({
      query: 'the-graphql-tag-query-string'
    }
  }
}
```

with `tslint.json` configuration:

```json
{
  "rules": {
    "indent": [true, "tabs", 2],
    "no-duplicate-variable": true,
    "no-unused-variable": [true]
  },
  "rulesDirectory": ["node_modules/tslint-eslint-rules/dist/rules"]
}
```

#### Actual behavior

This gives me the warning:

> ```
> tslint: src/providers/api.ts, line: 3
> 'MyInterface' is declared but never used.
> 
> L3:  import { MyInterface } from './interfaces';
> ```

#### Expected behavior

Passing the interface as a type to `watchQuery<T>` should see the declaration/import as being used.

**Comment by Abildtoft** [comment]
We're having the same issue, also TypeScript version: 2.3.4 and TSLint 5.7.0 (moving to 5.8.0 does not help this issue)

**Comment by ajafff** [comment]
IIRC there was a bug in older typescript versions that led to this error. Could you try upgrading to typescript@2.5?

**Comment by simonhaenisch** [comment]
upgraded to `typescript@2.6.1` and it's fixed (couldn't upgrade to 2.5 because I wasn't able to build my ionic project with that version). Thanks!

**Comment by sitepratico** [comment]
if it's used on template mark the variable as public to avoid showing this warning.
